instrumentation/action_pack/lib/opentelemetry/instrumentation/action_pack/handlers/action_controller.rb

```diff
@@ -44,35 +69,6 @@ def finish(_name, _id, payload)
           rescue StandardError => e
             OpenTelemetry.handle_error(exception: e)
           end
-
-          private
-
-          # Extracts the span name and attributes from the payload
-          #
-          # @param payload [Hash] the payload passed from ActiveSupport::Notifications
-          # @return [Array<String, Hash>] the span name and attributes
-          def to_span_name_and_attributes(payload)
-            request = payload[:request]
-            # It seems that there are cases in Rails functional tests where it bypasses the routing system and the `action_dispatch.route_uri_pattern` header not being set.
-            # Our Test suite executes the routing system so we are unable to recreate this error case.
-            # https://github.com/rails/rails/blob/747f85f200e7bb2c1a31b4e26e5a5655e2dc0cdc/actionpack/lib/action_dispatch/http/request.rb#L160
-            http_route = request.route_uri_pattern&.chomp('(.:format)') if request.respond_to?(:route_uri_pattern)
-
-            attributes = {
-              OpenTelemetry::SemanticConventions::Trace::CODE_NAMESPACE => String(payload[:controller]),
-              OpenTelemetry::SemanticConventions::Trace::CODE_FUNCTION => String(payload[:action])
-            }
-            attributes[OpenTelemetry::SemanticConventions::Trace::HTTP_ROUTE] = http_route if http_route
-            attributes[OpenTelemetry::SemanticConventions::Trace::HTTP_TARGET] = request.filtered_path if request.filtered_path != request.fullpath
-
-            if @span_naming == :semconv
-              return ["#{request.method} #{http_route}", attributes] if http_route
-
-              return ["#{request.method} /#{payload.dig(:params, :controller)}/#{payload.dig(:params, :action)}", attributes]
-            end
-
-            ["#{payload[:controller]}##{payload[:action]}", attributes]
```

Comment:
Inlined this method as it's not used elsewhere, and by doing so we can skip the array allocation used to return the result.